src/Microsoft.AspNet.Mvc.Core/UrlHelper.cs

```diff
@@ -99,12 +99,16 @@ private string GeneratePathFromRoute(IDictionary<string, object> values)
         protected virtual string GeneratePathFromRoute(string routeName, IDictionary<string, object> values)
         {
             var context = new VirtualPathContext(_httpContext, _ambientValues, values, routeName);
-            var path = _router.GetVirtualPath(context);
-            if (path == null)
+            var pathData = _router.GetVirtualPath(context);
+            if (pathData == null)
             {
                 return null;
             }
 
+            // VirtualPathData.VirtualPath returns string.Empty for null.
+            Debug.Assert(pathData.VirtualPath != null);
```

Comment:
This is going to become invalid with the next change (`PathString` is a struct), your call if you want to take it out now or then.